gstools/krige/base.py

```diff
@@ -28,9 +33,12 @@
 __all__ = ["Krige"]
 
 
+P_INV = {1: spl.pinv, 2: spl.pinv2, 3: spl.pinvh}
```

Comment:
A minor suggestion:

Why not let the methods take a function, instead of keys to this look-up-table, with a recommendation in the docstrings to use one of the following scipy-functions?
This would make things much more flexible, in case someone has a much better algorithm for computing the pseudo inverse. This will probably never happen, but at least things are not hidden behind magic numbers like 1, 2, 3.

Reply:
Why not both? I wanted to bring it in line with PyKrige (https://github.com/GeoStat-Framework/PyKrige/pull/151), where I need to pass this to a cython function and I don't want to pass a python function there. From my feeling, I like switches ;-)
But for UX reasons it would be nice to be able to have the ability to pass functions.

Reply:
"where I need to pass this to a cython function" - woah, don't go down that road!! I've been there before

https://stackoverflow.com/questions/17807359/cython-how-to-assign-a-c-function-to-a-python-cython-variable

Wow, that was actually my first question on SO.

But what do you mean with "Why not both?" - I think some of our function interfaces are already pretty... bloated, with arguments excluding each other and so on. - If you want to stay in line with PyKrige, than I'd suggest to simply stick to the lookup table. Maybe with a hint in the docstrings how this can be overwritten/ extended.

Reply:
With both I mean, that it can take a number or a callable. So you have a set of default methods and the possibility to just pass a callable right there:
```python
pseudo_inv_type=your_fancy_func
```
or
```python
pseudo_inv_type=1
```

Reply:
Okay, you like many options in the function arguments... then let's finally merge!